lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -2911,7 +2989,53 @@ LogicalResult SliceLibOp::verify() {
             DictionaryAttr::get(getContext())};                                \
   }
 
+#define ImplBinFloatingPointOpCellInterface(OpType)                            \
```

Comment:
We generally discourage macros. Could this be templated on `OpType` instead?

Reply:
+1, I see `ImplBinPipeOpCellInterface` is doing something similar and I'm not sure why 
 upon a quick glance. It might be OK to refactor both in a follow-up PR, or open an issue to do so and add a TODO.